libcst/metadata/tests/test_scope_provider.py

```diff
@@ -203,6 +203,39 @@ def test_dotted_import_access(self) -> None:
         self.assertEqual(list(scope_of_module["x.y"])[0].references, set())
         self.assertEqual(scope_of_module.accesses["x.y"], set())
 
+    def test_dotted_import_with_call_access(self) -> None:
+        m, scopes = get_scope_metadata_provider(
+            """
+            import os.path
+            os.path.join("A", "B").lower()
+            """
+        )
+        scope_of_module = scopes[m]
+        first_statement = ensure_type(m.body[1], cst.SimpleStatementLine)
+        attr = ensure_type(
+            ensure_type(
+                ensure_type(
+                    ensure_type(
+                        ensure_type(first_statement.body[0], cst.Expr).value, cst.Call
+                    ).func,
+                    cst.Attribute,
+                ).value,
+                cst.Call,
+            ).func,
+            cst.Attribute,
+        ).value
+        self.assertTrue("os.path" in scope_of_module)
+        self.assertTrue("os" in scope_of_module)
+
+        a_b_c_assignment = cast(Assignment, list(scope_of_module["os.path"])[0])
+        a_b_c_assignment_references = list(a_b_c_assignment.references)
+        self.assertNotEqual(len(a_b_c_assignment_references), 0)
+        a_b_c_access = a_b_c_assignment_references[0]
+        self.assertEqual(scope_of_module.accesses["os"], set())
+        self.assertEqual(scope_of_module.accesses["os.path"], {a_b_c_access})
+        self.assertEqual(scope_of_module.accesses["os.path.join"], set())
+        self.assertEqual(a_b_c_access.node, attr)
```

Comment:
The test case looks good.
Can we also add another test cases to just test the `_gen_dotted_names`? That can keep the test cases simpler and easier to add for future updates.

Reply:
Done! I added one test for that one